test/envtest/konnect_entities_kongconsumer_test.go

```diff
@@ -867,4 +869,94 @@ func TestKongConsumerSecretCredentials(t *testing.T) {
 		)
 	})
+
+	t.Run("HMAC", func(t *testing.T) {
+		consumerID := fmt.Sprintf("consumer-%d", rand.Int31n(1000))
+		username := fmt.Sprintf("user-secret-credentials-%d", rand.Int31n(1000))
+		cp := deploy.KonnectGatewayControlPlaneWithID(t, ctx, clientNamespaced, apiAuth)
+
+		t.Log("Setting up SDK expectations on KongConsumer creation")
+		sdk.ConsumersSDK.EXPECT().
+			CreateConsumer(mock.Anything, cp.GetKonnectStatus().GetKonnectID(),
+				mock.MatchedBy(func(input sdkkonnectcomp.ConsumerInput) bool {
+					return input.Username != nil && *input.Username == username
+				}),
+			).Return(&sdkkonnectops.CreateConsumerResponse{
+			Consumer: &sdkkonnectcomp.Consumer{
+				ID: lo.ToPtr(consumerID),
+			},
+		}, nil)
+
+		t.Log("Setting up SDK expectation on possibly updating KongConsumer ( due to asynchronous nature of updates between KongConsumer and KongConsumerGroup)")
+		sdk.ConsumersSDK.EXPECT().
+			UpsertConsumer(mock.Anything, mock.MatchedBy(func(r sdkkonnectops.UpsertConsumerRequest) bool {
+				return r.ConsumerID == consumerID
+			})).
+			Return(&sdkkonnectops.UpsertConsumerResponse{}, nil).
+			Maybe()
+
+		t.Log("Setting up SDK expectation on KongConsumerGroups listing")
+		sdk.ConsumerGroupSDK.EXPECT().
+			ListConsumerGroupsForConsumer(mock.Anything, sdkkonnectops.ListConsumerGroupsForConsumerRequest{
+				ConsumerID:     consumerID,
+				ControlPlaneID: cp.GetKonnectStatus().GetKonnectID(),
+			}).Return(&sdkkonnectops.ListConsumerGroupsForConsumerResponse{}, nil)
+
+		s := deploy.Secret(t, ctx, clientNamespaced,
+			map[string][]byte{
+				"username": []byte(username),
+				"secret":   []byte("hmac-secret"),
+			},
+			deploy.WithLabel("konghq.com/credential", konnect.KongCredentialTypeHMAC),
+		)
+
+		t.Log("Setting up SDK expectation on (managed) HMAC credentials creation")
+		sdk.KongCredentialsHMACSDK.EXPECT().
+			CreateHmacAuthWithConsumer(
+				mock.Anything,
+				mock.MatchedBy(
+					func(r sdkkonnectops.CreateHmacAuthWithConsumerRequest) bool {
+						return r.ControlPlaneID == cp.GetKonnectID() &&
+							r.HMACAuthWithoutParents.Username == username &&
+							*r.HMACAuthWithoutParents.Secret == "hmac-secret"
+					},
+				),
+			).
+			Return(
+				&sdkkonnectops.CreateHmacAuthWithConsumerResponse{
+					HMACAuth: &sdkkonnectcomp.HMACAuth{
+						ID: lo.ToPtr("hmac-auth-id"),
+					},
+				},
+				nil,
+			)
+		t.Log("Creating KongConsumer with ControlPlaneRef type=konnectID")
+		createdConsumer := deploy.KongConsumer(t, ctx, clientNamespaced, username,
+			deploy.WithKonnectIDControlPlaneRef(cp),
+			func(obj client.Object) {
+				consumer := obj.(*configurationv1.KongConsumer)
+				consumer.Credentials = []string{
+					s.Name,
+				}
+			},
+		)
+		t.Log("Waiting for KongConsumer to be programmed")
+		wConsumer := setupWatch[configurationv1.KongConsumerList](t, ctx, cl, client.InNamespace(ns.Name))
+		wHMAC := setupWatch[configurationv1alpha1.KongCredentialHMACList](t, ctx, cl, client.InNamespace(ns.Name))
```

Comment:
The watches should be set up before the object gets deployed so that we don't miss events

```suggestion
		t.Log("Creating KongConsumer with ControlPlaneRef type=konnectID")
		wConsumer := setupWatch[configurationv1.KongConsumerList](t, ctx, cl, client.InNamespace(ns.Name))
		wHMAC := setupWatch[configurationv1alpha1.KongCredentialHMACList](t, ctx, cl, client.InNamespace(ns.Name))
		createdConsumer := deploy.KongConsumer(t, ctx, clientNamespaced, username,
			deploy.WithKonnectIDControlPlaneRef(cp),
			func(obj client.Object) {
				consumer := obj.(*configurationv1.KongConsumer)
				consumer.Credentials = []string{
					s.Name,
				}
			},
		)
		t.Log("Waiting for KongConsumer to be programmed")
```